matchzoo/engine/base_model.py

```diff
@@ -105,9 +104,13 @@ def build(self):
 
     def compile(self):
         """Compile model for training."""
+        keras_metrics = []
+        for metric in self._params['task'].metrics:
+            if not isinstance(engine.parse_metric(metric), engine.BaseMetric):
+                keras_metrics.append(metric)
```

Comment:
I guess a better way to do is:
```python
activated_keras_metrics = []
for metric in self._params['task'].metrics:
    if keras.metrics.get(metric):
        activated_keras_metrics.append(metric)
```
take a look at [here](https://github.com/keras-team/keras/blob/df8186833fe47a6fd4b7dbc52fe12f9df73609fb/keras/metrics.py#L76), for the sake of parse.

Reply:
A metric is one of keras' metric, matchzoo's metric, or a erroneous metric (e.g. a metric with a typo). I propagate erroneous metrics to keras so keras will throw exceptions for me.